### ISSUE TITLE
Improve error messaging when typegen fails

### ISSUE THREAD

**Issue by nedsalk** [FuelLabs/fuels-ts#2833]
While doing #2826 I introduced some bad logic into typegen. After running `pnpm pretest` on `fuel-gauge` to build all the typegen types, typegen threw this rather unhelpful error:
```
Generating types..
Type not supported: [object Object]
```

Something better might be:
```
Generating types..
Error while generating types for abi: _/path/to/abi_
Trying to generate type: _the abi type name and id_
Type not supported: [object Object]
```

This can also help us when users encounter such issues.

**Comment by nedsalk** [comment]
There was an error on my part. After I rebuilt my `abi-typegen` package, this error was thrown:
```
Type not supported: (u32, u32)
```

This is much more helpful, but we can still add the message of which abi is the problematic one.

**Comment by petertonysmith94** [comment]
@nedsalk is this still relevant now that #2826 is closed?

**Comment by danielbate** [comment]
Resolved by #3032
